### PR TITLE
tickets/DM-17060

### PR DESCRIPTION
**Pull request by natelust**
Add Switch for setting Datatype name templates

### CODE REVIEW

File: python/lsst/pipe/base/configOverrides.py [hunk 1]
```diff
@@ -123,3 +140,16 @@ def applyTo(self, config):
                     # this can throw
                     value = eval(value, {})
                     setattr(obj, field[-1], value)
+            elif otype == 'namesDict':
+                try:
+                    parsedNamesDict = ast.literal_eval(override)
+                    if not isinstance(parsedNamesDict, dict):
+                        raise ValueError()
```
Comment (andy-slac) [inline python/lsst/pipe/base/configOverrides.py]: Some message to user would be useful here

File: python/lsst/pipe/base/configOverrides.py [hunk 2]
```diff
@@ -82,6 +84,21 @@ def addValueOverride(self, field, value):
...
+        a string in the format of key:value which will be used to format
+        fields in all of the nameTemplates found in a config object. I.E.
+        a nameDictString = {'input': deep} would be used to format a
+        nameTemplate of "{input}CoaddDatasetProduct".
+        Parameters
+        ----------
+        nameDictString : str
+            String formatted as a python dict used in formatting nameTemplates
+        """
+        self._overrides += [('namesDict', (nameDictString))]
```
Comment (andy-slac) [inline python/lsst/pipe/base/configOverrides.py]: Parens around `nameDictString` don't do anything

File: python/lsst/pipe/base/configOverrides.py [hunk 3]
```diff
@@ -82,6 +84,21 @@ def addValueOverride(self, field, value):
...
+
+        This method takes in a dictionary passed in on the command line as
+        a string in the format of key:value which will be used to format
+        fields in all of the nameTemplates found in a config object. I.E.
+        a nameDictString = {'input': deep} would be used to format a
+        nameTemplate of "{input}CoaddDatasetProduct".
+        Parameters
+        ----------
+        nameDictString : str
+            String formatted as a python dict used in formatting nameTemplates
```
Comment (andy-slac) [inline python/lsst/pipe/base/configOverrides.py]: I'd strongly prefer more type-safe interface here. Why no pass a `dict` here and construct that dict in `ctrl_mpexec` instead? This code is supposed to be more generic and reusable now and we should avoid depending on command-line-specific things here.

File: python/lsst/pipe/base/configOverrides.py (hunk 1 above)
Comment (timj) [reply python/lsst/pipe/base/configOverrides.py]: Python code isn't meant to raise `pexExceptions.RuntimeError`.

Comment (andy-slac) [reply python/lsst/pipe/base/configOverrides.py]: Regarding ValueError - I think it is more complicated. `literal_eval()` can certainly throw exceptions other than ValueError (SyntaxError is probably even more common). Also I'm not a big fan of chained exceptions as they can be confusing for end users (especially as in this case ValueError does not have any useful message) but I think we do not have any explicit policy against chaining. Still I think adding a message to ValueError here would be better.

**Pull request merged**